wandb/sdk/wandb_run.py

```diff
@@ -2477,13 +2477,21 @@ def _on_ready(self) -> None:
         self._telemetry_obj_active = True
         self._telemetry_flush()
 
+        self._detect_and_apply_job_inputs()
+
         # object is about to be returned to the user, don't let them modify it
         self._freeze()
 
         if not self._settings.resume:
             if os.path.exists(self._settings.resume_fname):
                 os.remove(self._settings.resume_fname)
 
+    def _detect_and_apply_job_inputs(self) -> None:
+        """If the user has staged launch inputs, apply them to the run."""
+        from wandb.sdk.launch.inputs.internal import StagedLaunchInputs
+
+        StagedLaunchInputs().apply(self)
+
```

Comment:
do you need to check that launch is not disabled?

Reply:
also it doesn't seem like it needs to be a method on the run object, could be just a general function that gets run object as input?

Reply:
but looking at the code i see we have many patterns like this... probably would refactor all of it when we re-write the client

Reply:
seems to me that you are missing handling the case when the user disables launch